semihost/open.c

```diff
@@ -71,6 +71,8 @@ open(const char *pathname, int flags, ...)
 			semiflags = SH_OPEN_W_PLUS_B;	/* 'wb+' */
 		else if (flags & O_APPEND)
 			semiflags = SH_OPEN_A_PLUS_B;	/* 'ab+' */
+		else if (flags & O_CREAT)
```

Comment:
This isn't right either -- using 'ab+' will put the initial file position at the end of the file and cause new writes
to append to the file. I couldn't find a mode that will open the file for reading and writing, creating it if necessary, without placing the file in append mode. Maybe this requires a sequence of two operations? First attempt to use 'rb+', and if that fails, use 'wb+'?